torch/ao/sparsity/sparsifier/weight_norm_sparsifier.py

```diff
@@ -54,13 +54,22 @@ def __init__(self,
 
     def update_mask(self, layer, sparsity_level, sparse_block_shape,
                     zeros_per_block, **kwargs):
-        if zeros_per_block != reduce((lambda x, y: x * y), sparse_block_shape):
-            raise NotImplementedError('Partial block sparsity is not yet there')
+        values_per_block = reduce((lambda x, y: x * y), sparse_block_shape)
+        if zeros_per_block > values_per_block:
+            raise ValueError("Number of zeros per block cannot be more than "
+                             "the total number of elements in that block.")
+        elif sparsity_level <= 0.0:
```

Comment:
nit: probably just `if` since this check is not too relevant to the previous one?
I'm assuming sparsity_level is a percentage of sparsity and when it is <= 0 it means it's not sparsified and when it is >= 1.0 the tensor should be fully sparsified? could you add this to the docstring above?

also does the concept of sparsity_level and values_per_block overlap?

Reply:
qq, is there a difference between this line and L70?  From reading the code, seems like if sparsity_level <= 0.0 then L63 would return and L70 would not be reached...did I get that right?

Reply:
> also does the concept of sparsity_level and values_per_block overlap?

Not directly. We define the `sparsity_level` as "ratio of blocks that are sparse". However, `values_per_block` defines the number of elements in a block (basically shape[0] * shape[1] * ...). The latter is then used to compute the sparsity level within each block. So, for example, if `zeros_per_block = 2` and `values_per_block = 4`, that would mean that block is expected to be 50% sparse. I will have a brief about the API early next week

Reply:
> qq, is there a difference between this line and L70? From reading the code, seems like if sparsity_level <= 0.0 then L63 would return and L70 would not be reached...did I get that right?

Oh yea -- good catch, there is some code duplication going on here. Will recheck it